src/main/java/duke/tasks/TaskManager.java

```diff
@@ -0,0 +1,151 @@
+        int stringLength = args.length();
+        int taskIndex = Integer.parseInt(args.substring(stringLength - 1));
+
+        System.out.println(LINE_SEPARATOR);
+        System.out.println("Got it! This task was removed:" + System.lineSeparator()
+                + taskList.get(taskIndex - 1).toString());
+
+        // if deleted task was not marked as done, decrease numberOfTasksUndone
+        if (!taskList.get(taskIndex - 1).getStatusIcon().equals("X")) {
+            numberOfTasksUndone--;
+        }
+        taskList.remove(taskIndex - 1);
+
+        printTaskNumberMessage(taskList.size());
+    }
+
+    /**
+     * Store to do tasks in a list.
+     *
+     * @param args  the item after the command the user inputs
+     **/
+    public static void addToDoTaskToList(String args) {
+
+        ToDo t = new ToDo(args);
+        taskList.add(t);
+        numberOfTasksUndone++;
+        System.out.println(LINE_SEPARATOR);
+        System.out.println("added: " + args);
+        printTaskNumberMessage(taskList.size());
+    }
+
+    /**
+     * Store deadline tasks in a list.
+     *
+     * @param args  the item after the command the user inputs
+     **/
+    public static void addDeadlineTaskToList(String args) {
+        String description = args.substring(0, args.indexOf(BY_DIVIDER)).trim();
+        String time = args.substring(args.indexOf(BY_DIVIDER) + 4);
+        Deadline t = new Deadline(description, time);
+        taskList.add(t);
+        numberOfTasksUndone++;
+        System.out.println(LINE_SEPARATOR);
+        System.out.println("added: " + description);
+        printTaskNumberMessage(taskList.size());
+    }
+
+    /**
+     * Store event tasks in a list.
+     *
+     * @param args  the item after the command the user inputs
+     **/
+    public static void addEventTaskToList(String args) {
+        String description = args.substring(0, args.indexOf(AT_DIVIDER)).trim();
+        String time = args.substring(args.indexOf(AT_DIVIDER) + 4);
+        Event t = new Event(description, time);
+        taskList.add(t);
+        numberOfTasksUndone++;
+        System.out.println(LINE_SEPARATOR);
+        System.out.println("added: " + description);
+        printTaskNumberMessage(taskList.size());
+    }
+
+    /**
+     * Mark a task as done and prints done message
+     *
+     * @param args the string the user inputs
+     **/
+    public static void markTaskAsDone(String args) {
+        int stringLength = args.length();
+        int doneTaskNumber = Integer.parseInt(args.substring(stringLength - 1));
+        taskList.get(doneTaskNumber - 1).markAsDone();
+        numberOfTasksUndone--;
+
+        System.out.println(LINE_SEPARATOR);
+        System.out.println("Good job! This task is marked as done:");
+        System.out.println(taskList.get(doneTaskNumber - 1).toString());
+        System.out.println("Now you have " + numberOfTasksUndone + " tasks undone");
+        System.out.println(LINE_SEPARATOR);
+    }
```

Comment:
Perhaps you could use inheritance here and add have ToDo, Event and Deadline classes have instance variable `isDone` and then you can just include this functionality for each one of them